File: tests/fixtures/check_then_act/ts/revalidated_fixture.ts
// The state is read and checked again after the await, so the first check is not what the write relies on.
export async function apply(repo: any, id: string) {
  const first = repo.get(id);
  if (!first) return;
  await repo.refresh(id);
  const current = repo.get(id);
  if (!current || current.done) return;
  repo.update(id, { done: true });
}

File: tests/fixtures/check_then_act/py/revalidated_fixture.py
# The state is read and checked again after the await, so the first check is not what the write relies on.
async def apply(repo, item_id):
    first = repo.get(item_id)
    if not first:
        return
    await repo.refresh(item_id)
    current = repo.get(item_id)
    if not current or current.done:
        return
    repo.update(item_id, done=True)
